homeassistant/components/media_player/mediaroom.py

```diff
@@ -152,50 +188,114 @@ def media_content_type(self):
         """Return the content type of current playing media."""
         return MEDIA_TYPE_CHANNEL
 
-    def turn_on(self):
+    @property
+    def media_channel(self):
+        """Channel currently playing."""
+        return self._channel
+
+    async def async_turn_on(self):
         """Turn on the receiver."""
-        self.stb.send_cmd('Power')
-        self._state = STATE_ON
+        from pymediaroom import PyMediaroomError
+        if self._optimistic:
+            return
+
+        try:
+            self.set_state(await self.stb.turn_on())
+        except PyMediaroomError:
+            self._state = STATE_UNAVAILABLE
```

Comment:
All these have to set `self._available = False` and the property `available` will need to return `self._available`